### ISSUE TITLE
Release tag in master

### ISSUE THREAD

**Issue by xkszltl** [microsoft/onnxruntime#929]
We are using git describe to pick up the version number from release tag.
This works fine but not on master.
The latest tag on master is v0.1.4

**System information**
- ONNX Runtime version (you are using): master

**Describe the solution you'd like**

Merge tagged release branch back to master, even with something like `git checkout --ours *` to take everything from master branch only in the "Merge" commit.

All we need is an git edge in history.

**Comment by pranavsharma** [comment]
We're in the process of cutting a release branch for 0.1.4. It has not been cut yet. It'll be ready early next week. Thanks for your patience.

**Comment by xkszltl** [comment]
@pranavsharma 
Sorry I don't really understand the word.
What does "cutting a release branch for 0.1.4" means?

> We're in the process of cutting a release branch for 0.1.4. It has not been cut yet. It'll be ready early next week. Thanks for your patience.

**Comment by raymondxyang** [comment]
We cut the branch for the release for potential patches. It is requested by some partner teams.

**Comment by xkszltl** [comment]
Oh that's probably not related to this?

What I'm asking is to run a PR from v0.3.1 to master, so that "git describe" can get a tag on master

**Comment by pranavsharma** [comment]
Can you please elaborate? What’s the objective?

**Comment by xkszltl** [comment]
TLDR: We want to have an easy and general way to get a version number, by using `git describe`.

`git describe` command returns the latest annotated tag on the branch + how many commits ahead of it, in the form of `<tag>-<# commits ahead>-<hash>`.
This provides information similar to `/VERSION_NUMBER` but more git friendly.

Since master should be considered as a version ahead of any release, it would be better to also express this relation in git.

**Comment by xkszltl** [comment]
I also noticed there're cherry-pick commit from release branch back to master.
Merge commit may be better than cherry-pick.
If there's certain "release-only" commit, you can discard those changes when merging back to master.
Comparing to cherry-pick, this practice explicitly tells every stories in the history, and keep it "linear" for the long run.

**Comment by pranavsharma** [comment]
We don't plan to work on this as it's not one of our top priorities. Feel free to re-open if this is a blocker for your release. Thanks for your interest.
